listenbrainz/listenstore/timescale_listenstore.py

```diff
@@ -454,6 +454,111 @@ def fetch_recent_listens_for_users(self, users, min_ts: datetime = None, max_ts:
                 ))
         return listens
 
+
+    def fetch_recent_listens_for_users_new(self, users, min_ts: datetime, max_ts: datetime, limit=25):
+        """ Fetch recent listens for a list of users.
+
+            users: A list containing the users for which you'd like to retrieve recent listens.
+            min_ts: Only return listens with listened_at after this timestamp, required.
+            max_ts: Only return listens with listened_at before this timestamp, required.
+        """
+
+        user_id_map = {user["id"]: user["musicbrainz_id"] for user in users}
+
+        args = {"user_ids": tuple(user_id_map.keys()), "limit": max(limit, 100)}
+
+        # min_ts and max_ts must exist.
+        if min_ts and max_ts:
+            args["min_ts"] = min_ts
+            args["max_ts"] = max_ts
+        else: 
+            raise ValueError()
+
+        filters = """user_id IN :user_ids 
```

Comment:
Since the fliters are always the same, just write in the query directly.

Reply:
Alright!